msi/createmsi.py

```diff
@@ -116,40 +75,104 @@ def build_dist(self):
+        product = ET.SubElement(self.root, 'Product', {
+            'Name': self.product_name,
+            'Manufacturer': 'The Meson Development Team',
+            'Id': self.guid,
+            'UpgradeCode': self.update_guid,
+            'Language': '1033',
+            'Codepage':  '1252',
+            'Version': self.version,
+            })
+
+        ET.SubElement(product, 'Package',  {
+            'Id': '*',
+            'Keywords': 'Installer',
+            'Description': 'Meson %s installer' % self.version,
+            'Comments': 'Meson is a high performance build system',
+            'Manufacturer': 'The Meson Development Team',
+            'InstallerVersion': '100',
+            'Languages': '1033',
+            'Compressed': 'yes',
+            'SummaryCodepage': '1252',
+            })
+        ET.SubElement(product, 'Media', {
+            'Id': '1',
+            'Cabinet': 'meson.cab',
+            'EmbedCab': 'yes',
+            })
+        targetdir = ET.SubElement(product, 'Directory', {
+            'Id': 'TARGETDIR',
+            'Name': 'SourceDir',
+            })
+        progfiledir = ET.SubElement(targetdir, 'Directory', {
+            'Id' : self.progfile_dir,
+            })
+        installdir = ET.SubElement(progfiledir, 'Directory', {
+            'Id': 'INSTALLDIR',
+            'Name': 'Meson'})
+
+        ET.SubElement(product, 'Property', {
+            'Id': 'WIXUI_INSTALLDIR',
+            'Value': 'INSTALLDIR',
+            })
+        ET.SubElement(product, 'UIRef', {
+            'Id': 'WixUI_InstallDir',
+            })
         assert(os.path.isdir(self.staging_dir))
-        comp_ref_xml = ''
         nodes = {}
-        with open(self.main_xml, 'w') as ofile:
-            for root, dirs, files in os.walk(self.staging_dir):
-                cur_node = Node(dirs, files)
-                nodes[root] = cur_node
-            ofile.write(xml_templ % (self.guid, self.update_guid, self.version, self.version,
-                                     self.platform_str, self.progfile_dir))
-            self.component_num = 0
-            self.create_xml(nodes, ofile, self.staging_dir)
-            for i in range(self.component_num):
-                comp_ref_xml += comp_ref_templ % ('ApplicationFiles%d' % i)
-            ofile.write(xml_footer_templ % comp_ref_xml)
-
-    def create_xml(self, nodes, ofile, root):
-        cur_node = nodes[root]
+        for root, dirs, files in os.walk(self.staging_dir):
+            cur_node = Node(dirs, files)
+            nodes[root] = cur_node
+        self.component_num = 0
+        self.create_xml(nodes, self.staging_dir, installdir)
+        feature = ET.SubElement(product, 'Feature', {
+            'Id': 'DefaultFeature',
+            'Level': '1',
+            })
+        
+        for i in range(self.component_num):
+            ET.SubElement(feature, 'ComponentRef', {
+                'Id': 'ApplicationFiles%d' % i,
+                })
```

Comment:
__[flake8]__
* [E123] closing bracket does not match indentation of opening bracket's line

<!-- Comment by [SideCI](https://sideci.com) -->